client/src/components/DualRangeSlider.js

```diff
@@ -1,55 +1,66 @@
-import React from "react";
+import React, { useState } from "react";
 import { Range } from "react-range";
 import "./DualRangeSlider.css";
 
-const DualRangeSlider = ({ selectedRange, onChangeRange }) => {
+const DualRangeSlider = ({ onChangeRange, isNightMode }) => {
 	const min = 1600;
-	const max = 2023;
-	const step = 50;
+	const max = 2050;
+	const step = 150;
+
+	const labels = ["Pre-colonial", "Pre-civil war", "Post civil war", "Present"];
+
+	const [values, setValues] = useState([min, max]);
 
 	const handleChange = (newValues) => {
+		setValues(newValues);
 		onChangeRange(newValues);
 	};
 
-	const railLabels = [1700, 1800, 1900];
+	const getLabelForValue = (value) => {
+		// work out index of current label based on value and step count to be used in title
+		const index = Math.floor((value - min) / step);
+		return labels[index];
+	};
 
 	return (
 		<div className="dual-range-container">
 			<h3 className="range-title">
-				Range: {selectedRange[0]} - {selectedRange[1]}
+				Range: {getLabelForValue(values[0])} - {getLabelForValue(values[1])}
 			</h3>
 			<div className="range-wrapper">
 				<Range
 					step={step}
 					min={min}
 					max={max}
-					values={selectedRange}
+					values={values}
 					onChange={handleChange}
 					renderTrack={({ props, children }) => (
```

Comment:
I'd consider moving `renderThumb`, `renderTrack` and `renderMark` into functions defined above this component so you can just say `renderMark={renderMark}`. It's a bit more readable